scribejava-core/src/main/java/com/github/scribejava/core/oauth/OAuth20Service.java

```diff
@@ -34,7 +38,18 @@ public OAuth20Service(DefaultApi20 api, OAuthConfig config) {
 
     //sync version, protected to facilitate mocking
     protected OAuth2AccessToken sendAccessTokenRequestSync(OAuthRequest request) throws IOException {
-        return api.getAccessTokenExtractor().extract(request.send().getBody());
+        final Response response = request.send();
+        final TokenExtractor<OAuth2AccessToken> extractor = api.getAccessTokenExtractor();
+        if (response.getCode() != 200) {
+            if (extractor instanceof OAuth2AccessTokenJsonExtractor) {
+                final OAuth2AccessTokenJsonExtractor jsonExtractor = (OAuth2AccessTokenJsonExtractor) extractor;
+                throw new OAuthRequestException("Get Access token error",
+                        jsonExtractor.extractError(response.getBody()));
+            } else {
+                throw new OAuthException("Get Access token error: " + response.getBody());
```

Comment:
IMO it's better to hide all this logic in the single extractor method

like this
```
public interface TokenExtractor<T extends Token, E extends OAuthException> {

    /**
     * Extracts the concrete type of token from the contents of an Http Response
     *
     * @param response the contents of the response
     * @return OAuth access token
     */
    T extract(String response) throws E;
}

public class OAuth2AccessTokenJsonExtractor implements TokenExtractor<OAuth2AccessToken, OAuth2ErrorResponse> {
...
```
?